### PR TITLE
Fix/axios refactor

### PR DESCRIPTION
**Pull request by ajhollid**
Axios requests are made all over the app and are difficult to track down for refactoring.  This PR aims to move all axios requests to the axios instance for ease of maintenance

- [x] Move `uptimeMonitorSlice` operations
- [x] Move `pageSpeedMonitorSlice` operations
- [x] Move auth operations 
- [x] Move all non-redux operations
- [x] Refactor Axios util into a class based service
- [x] JSdocs

This refactor moves all of our Axios requests to one class based service rather than having them spread throughout the application.

Maintainability should be significantly improved

### CODE REVIEW

File: Server/controllers/authController.js [hunk 1]
```diff
@@ -450,6 +450,8 @@ const deleteUserController = async (req, res, next) => {
     }
 
     // 1. Find all the monitors associated with the user id
+    console.log("wtf");
```
Comment (mertssmnoglu) [inline Server/controllers/authController.js:453]: There is unwanted console.log here

Comment (gorkem-bwl) [reply Server/controllers/authController.js]: Lol

File: Client/src/Utils/NetworkService.js [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+      (error) => {
+        console.error(error);
+        if (error.response && error.response.status === 401) {
+          console.log("Invalid token revoked");
+          networkService;
+        }
+        return Promise.reject(error);
+      }
+    );
+  }
```
Comment (mertssmnoglu) [inline Client/src/Utils/NetworkService.js:20]: Adding `this.defaultHeaders` will help us to extend new methods. Most(or all) endpoints work with `Content-Type: application/json` header.

**Review by mertssmnoglu — APPROVED** [review]
Looks nice to me

File: Client/src/Pages/PageSpeed/Details/index.jsx [hunk 3]
```diff
@@ -186,15 +186,15 @@ const PageSpeedDetails = () => {
...
-            },
-          }
+        const res = await networkService.getStatsByMonitorId(
+          authToken,
+          monitorId,
+          "desc",
+          50,
+          null,
+          null,
+          null
```
[the remark this answers is not in the document]
Comment (ajhollid) [reply Client/src/Pages/PageSpeed/Details/index.jsx:208]: Passing a standardized config object is a nice idea, we can add that to the next revision

**Review by danielcj2 — APPROVED** [review]
Nice work!

**Pull request merged**